src/main/java/eu/dissco/backend/controller/UserController.java

```diff
@@ -0,0 +1,124 @@
+package eu.dissco.backend.controller;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import eu.dissco.backend.domain.JsonApiLinks;
+import eu.dissco.backend.domain.JsonApiWrapper;
+import eu.dissco.backend.exceptions.ConflictException;
+import eu.dissco.backend.exceptions.ForbiddenException;
+import eu.dissco.backend.exceptions.NotFoundException;
+import eu.dissco.backend.service.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.keycloak.KeycloakPrincipal;
+import org.keycloak.KeycloakSecurityContext;
+import org.keycloak.representations.AccessToken;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@Slf4j
+@RestControllerAdvice
+@RestController
+@RequestMapping("/api/v1/users")
+@RequiredArgsConstructor
+public class UserController {
+
+  private static final String SELF_LINK = "https://sandbox.dissco.tech/api/v1/users/";
+
+  private final UserService service;
+
+  private static void checkAuthorisation(String tokenId, String id) throws ForbiddenException {
+    if (!tokenId.equals(id)) {
+      throw new ForbiddenException("User: " + tokenId + " is not allowed to perform this action");
+    }
+  }
+
+  @PreAuthorize("isAuthenticated()")
+  @ResponseStatus(HttpStatus.CREATED)
+  @PostMapping
+  public ResponseEntity<JsonApiWrapper> createNewUser(Authentication authentication,
+      @RequestBody JsonApiWrapper request)
+      throws JsonProcessingException, ConflictException, ForbiddenException {
+    var tokenId = getNameFromToken(authentication);
+    log.info("User: {} has requested to update user information of: {}",
+        tokenId, request.data().id());
+    checkAuthorisation(tokenId, request.data().id());
```

Comment:
Can you explain how comparing the user tokenId to request.data().id ensures the user is authorized? Could the request id not be manipulated?

Reply:
A different user could use the userId of another user in the request and then delete that user :smiling_face_with_tear: 
So let's say I have your userId (which are public) and send a request to the server with a delete for your userId. That would be totally acceptable if we didn't do this check as you are a authorized user. So that's where the check comes in.
- Create/Update/Delete a user can only do for itself (although delete is a bit strange)
- Find all authorized users can do (might become public later, not sure about that)

Reply:
Example:
DELETE -> `localhost:8080/api/v1/users/e2befba6-9324-4bb4-9f41-d7dfae4a44b0`
For my user with the bearer token for user `adf294ba-bb03-4962-8042-a37f1648458e`